src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -158,11 +165,12 @@ class DeepDiveLogParser extends JavaTokenParsers {
 
   def operator = "||" | "+" | "-" | "*" | "/" | "&"
   def typeOperator = "::"
-  val aggregationFunctions = Set("MAX", "SUM", "MIN", "ARRAY_ACCUM", "ARRAY_AGG")
+  val aggregationFunctions = Set("MAX", "SUM", "MIN", "ARRAY_ACCUM", "ARRAY_AGG", "COUNT")
 
-  // expressions
+  // expression
   def expr : Parser[Expr] =
-    ( lexpr ~ operator ~ expr ^^ { case (lhs ~ op ~ rhs) => BinaryOpExpr(lhs, op, rhs) }
+    ( "_" ^^ { case _ => Placeholder() }
```

Comment:
I wished we first separated patterns from expr before supporting underscore wildcards.  If underscore is used in the head, it doesn't make sense and should be ideally a parse error.

Without complex data type support, the cases for pattern type could be as simple as a wildcard or a binding to a variable or constraint by an expression.  Introducing that may actually make the unification code more clear.  How hard does this sound?  We could do this in a separate PR if it's too much.